L1Trigger/Phase2L1ParticleFlow/interface/L1NNTauProducer.hh

```diff
@@ -0,0 +1,41 @@
+#ifndef L1TRIGGER_PHASE2L1PARTICLEFLOW_L1NNTAU_H
+#define L1TRIGGER_PHASE2L1PARTICLEFLOW_L1NNTAU_H
+
+#include <iostream>
+#include <vector>
+#include <TLorentzVector.h>
+
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+
+#include "DataFormats/L1TParticleFlow/interface/PFTau.h"
+#include "DataFormats/L1TParticleFlow/interface/PFCandidate.h"
+#include "L1Trigger/Phase2L1ParticleFlow/interface/TauNNId.h"
+
+using namespace l1t;
+
+class L1NNTauProducer : public edm::stream::EDProducer<> {
+public:
+  explicit L1NNTauProducer(const edm::ParameterSet &);
+  ~L1NNTauProducer();
+
+private:
+  TauNNId *fTauNNId;
+  void addTau(l1t::PFCandidate &iCand,
+              const l1t::PFCandidateCollection &iParts,
+              std::unique_ptr<PFTauCollection> &outputTaus);
+  float deltaR(auto iPart1, auto iPart2);
```

Comment:
use existing https://github.com/cms-sw/cmssw/blob/master/DataFormats/Math/interface/deltaR.h

Reply:
(also, abbreviated function templates using auto are a C++20 feature, not supported in CMSSW yet, hence the compilation error in the PR test)